config/schemas/6/rapidast_schema.json

```diff
@@ -0,0 +1,605 @@
+                                    "type": "string"
+                                }
+                            },
+                            "required": [
+                                "proxyHost",
+                                "proxyPort"
+                            ]
+                        },
+                        "authentication": {
+                            "type": "object",
+                            "properties": {
+                                "type": {
+                                    "type": "string",
+                                    "enum": [
+                                        "oauth2_rtoken",
+                                        "http_basic",
+                                        "http_header",
+                                        "cookie",
+                                        "browser"
+                                    ]
+                                },
+                                "parameters": {
+                                    "type": "object",
+                                    "properties": {}
+                                }
+                            },
+                            "required": [
+                                "type",
+                                "parameters"
+                            ],
+                            "allOf": [
+                                {
+                                    "if": {
+                                        "properties": {
+                                            "type": {
+                                                "const": "oauth2_rtoken"
+                                            }
+                                        }
+                                    },
+                                    "then": {
+                                        "properties": {
+                                            "parameters": {
+                                                "properties": {
+                                                    "token_endpoint": {
+                                                        "type": "string"
+                                                    },
+                                                    "client_id": {
+                                                        "default": "cloud-services",
+                                                        "type": "string"
+                                                    },
+                                                    "rtoken": {
+                                                        "type": "string"
+                                                    },
+                                                    "preauth": {
+                                                        "type": "string"
+                                                    }
+                                                },
+                                                "required": [
```

Comment:
required parameters for `oauth2_rtoken` authentication

Reply:
client_id and token_endpoint are required.

Reply:
'cloud-services' is just one example. every user may have different client_id. If it helps to avoid confusion, removing the code defaulting could be done too IMO.

Reply:
Alright, I'll add both parameters to the required fields. Regarding the code modification, I agree that this will improve consistency. I can make those changes as well, but we should include a note in the release to highlight that this change is not backward compatible. Is there a document where I can document this for the future release?

Reply:
Thanks. I've tagged you in a temporary doc.

Reply:
Done